### ISSUE TITLE
high cpu when connected to IB data source

### ISSUE THREAD

**Issue by rterbush** [mementum/backtrader#250]
As mentioned, have been able to replicate this issue on both python3 and python2. Running on python2 and looking at a few things in gdb, I'm just attaching to a running process consuming a little over 100% cpu in an attempt to see what it is so focused on... I see the following. Just capturing this data for possible future use in tracking this down for now.

Not sure this tells us much other than the obvious that Cerebro spends a lot of time with data feeds...

```
(gdb) py-bt
Traceback (most recent call first):
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/feeds/ibdata.py", line 202, in _timeoffset
    def _timeoffset(self):
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 55, in __init__
    _dtime = datetime.utcnow() + data._timeoffset()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 138, in _checkbarover
    chkdata = DTFaker(data, forcedata) if fromcheck else data
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 506, in __call__
    forcedata=forcedata)):
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 315, in check
    return self(data, fromcheck=True, forcedata=_forcedata)
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/feed.py", line 421, in _check
    ff.check(self, _forcedata=forcedata, *fargs, **fkwargs)
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/cerebro.py", line 1233, in _runnext
    data._check()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/cerebro.py", line 940, in runstrategies
    self._runnext(runstrats)
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/cerebro.py", line 810, in run
    runstrat = self.runstrategies(iterstrat)
```
```
(gdb) py-print self
local 'self' = <IBData(tick_openinterest=<float at remote 0x8104c57b0>, _qcheck=<float at remote 0x805491958>, sessionend=<float at remote 0x81122af88>, cashtype=False, tick_high=<float at remote 0x8104c5738>, _state=2, _statelivereconn=True, plotinfo=<AutoInfoClass_pi_LineSeries_pi_DataSeries_pi_OHLC_pi_OHLCDateTime_pi_AbstractDataBase_pi_DataBase_pi_IBData(plot=False, plotmaster=None, plotind=True) at remote 0x8131495a0>, plotlines=<AutoInfoClass_pl_LineSeries_pl_DataSeries_pl_OHLC_pl_OHLCDateTime_pl_AbstractDataBase_pl_DataBase_pl_IBData(high=<AutoInfoClass_pl_OHLC_high at remote 0x813149f40>, datetime=<AutoInfoClass_pl_OHLCDateTime_datetime at remote 0x8131511b0>, volume=<AutoInfoClass_pl_OHLC_volume at remote 0x813149ed0>, low=<AutoInfoClass_pl_OHLC_low at remote 0x813149fb0>, close=<AutoInfoClass_pl_OHLC_close at remote 0x813151060>, openinterest=<AutoInfoClass_pl_OHLC_openinterest at remote 0x8131510d0>, open=<AutoInfoClass_pl_OHLC_open at remote 0x813151140>) at remote 0x813149760>, ib=<IBStore(_event_managed_accoun...(truncated)
```
```
(gdb) t a a py-bt

Thread 2 (LWP 103286 of process 36082):
Traceback (most recent call first):
  Waiting for the GIL
  <built-in method recv of _socket.socket object at remote 0x81314f8c8>
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/ib/lib/__init__.py", line 125, in readByte
    return unpack('!b', self.recv(1))[0]
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/ib/ext/EReader.py", line 895, in readStr
    c = self.m_dis.readByte()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/ib/ext/EReader.py", line 910, in readInt
    strval = self.readStr()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/ib/ext/EReader.py", line 156, in processMsg
    tickerId = self.readInt()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/ib/ext/EReader.py", line 107, in run
    while not self.isInterrupted() and self.processMsg(self.readInt()):
  File "/usr/local/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()

Thread 1 (LWP 102050 of process 36082):
Traceback (most recent call first):
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/feeds/ibdata.py", line 202, in _timeoffset
    def _timeoffset(self):
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 55, in __init__
    _dtime = datetime.utcnow() + data._timeoffset()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 138, in _checkbarover
    chkdata = DTFaker(data, forcedata) if fromcheck else data
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 506, in __call__
    forcedata=forcedata)):
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/resamplerfilter.py", line 315, in check
    return self(data, fromcheck=True, forcedata=_forcedata)
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/feed.py", line 421, in _check
    ff.check(self, _forcedata=forcedata, *fargs, **fkwargs)
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/cerebro.py", line 1233, in _runnext
    data._check()
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/cerebro.py", line 940, in runstrategies
    self._runnext(runstrats)
  File "/home/inmate/.virtualenvs/backtrader2/lib/python2.7/site-packages/backtrader/cerebro.py", line 810, in run
    runstrat = self.runstrategies(iterstrat)
```

**Comment by rterbush** [comment]
As mentioned in community, this appears to be related to resampling another IB feed to a different compression level. ie. 

```python
        data0 = ibstore.getdata(dataname, timeframe=bt.TimeFrame.Minutes, compression=1)
        cerebro.resampledata(data0, name='ES', timeframe=bt.TimeFrame.Minutes,
                             compression=1)

        cerebro.resampledata(data0, name='ES-daily', timeframe=bt.TimeFrame.Days, compression=1)
```

**Comment by mementum** [comment]
Fix in 8eb95697a711754d2217d9cf2a38de63779cf52b
